maze.py

```diff
@@ -0,0 +1,162 @@
+    # Fill screen with blank (off pixels)
+    screen.pixel_buffer = [[EMPTY_COLOR for _ in range(SCREEN_SIZE)] for _ in range(SCREEN_SIZE)]
+
+    # Draw player trail
+    if show_trail:
+        for pos in player.trail:
+            draw_cell(screen, pos[1], pos[0], TRAIL_COLOR)
+
+    # Draw player
+    draw_cell(screen, player.y, player.x, TRAIL_COLOR)
+    screen.set_image(screen.pixel_buffer)
+
+
+def draw_cell(screen, row, col, color):
+    if screen.is_drawable(row, col):
+        screen.pixel_buffer[row][col] = color
+
+def check_trail(player):
+    image = [[0.0 for c in range(SCREEN_SIZE)] for r in range(SCREEN_SIZE)]
+
+    loosing_string = String(image, (1, 18), 45, "Y o u  L o s t")
+    winning_string = String(image, (1, 18), 45, "L e v e l C l e a r e d", cooldown=0.03, timeout=0)
+
+    current_length = len(player.trail_directions)
+    if player.trail_directions != level_directions[:current_length]:
+        return loosing_string, True  # Return "You Lost" and set game_over as True
+    elif player.trail_directions == level_directions:
+        return winning_string, True   # Return "Level Cleared" and set game_over as True
+    return None, False  # No game over yet
+
+# Function to hide trail after TRAIL_TIME and reset the player's position
+def hide_trail(player):
+    player.reset()  # Reset the player's position and trail
+    global game_started
+    game_started = True
+
+def display_message_until_escape(string_object, screen):
+    """Display the string message until the player presses ESC to exit."""
+    image = [[0.0 for c in range(SCREEN_SIZE)] for r in range(SCREEN_SIZE)]
+    string_object.image = image
+
+    running = True
+    while running:
+        # Keep showing the message until ESC is pressed
+        string_object.scroll(inverted=False)
+        screen.set_image(image)
+        
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                pygame.quit()
+                sys.exit()
+            elif event.type == pygame.KEYDOWN:
+                if event.key == pygame.K_ESCAPE:
+                    running = False
+
+        pygame.time.Clock().tick(30)
+
+def game_loop():
+    # Define starting point at the very bottom middle of the cross
+    start_x = SCREEN_SIZE // 2  # Middle of the screen horizontally
+    start_y = SCREEN_SIZE - 1   # Very bottom of the screen
+    player = Player(start_x, start_y)
+
+    screen = PharmaScreen(True)
+
+    # Start a timer to hide the trail after TRAIL_TIME and reset the player
+    threading.Timer(TRAIL_TIME, lambda: hide_trail(player)).start()
+
+    for direction in level_directions:
+        player.move(direction)
+
+    global game_over
+
+    running = True
+    while running:
+        for event in pygame.event.get():
+            if event.type == pygame.QUIT:
+                pygame.quit()
+                sys.exit()
+            elif event.type == pygame.KEYDOWN:
+                if event.key == pygame.K_LEFT:
```

Comment:
I think using [match case](https://peps.python.org/pep-0636/) would be slightly better here